### PR TITLE
Components client side storage

### PR DESCRIPTION
**Pull request by babycourageous**
Adds Client Side Storage recipe

### CODE REVIEW

File: components.md [hunk 1]
```diff
@@ -292,6 +293,129 @@ _to be written_
...
+  <h1>Svelte with Basic Local Storage</h1>
+  <input type="text" bind:value />
+  <button on:click="{saveToLocal}">Save to Local</button>
+  <p>{value}</p>
+</div>
+```
+
+Type something in the `input` field and hit the `button`. When you refresh the page, voila the data has persisted and is used as the default value for the `value` prop.
+
+### More Complex Values
```
Comment (dasDaniel) [inline components.md:321]: this seems a little long for a _"recipe"_

File: components.md [hunk 2]
```diff
@@ -292,6 +293,129 @@ _to be written_
...
+The rest is pretty standard Svelte Todo app business.
+
+### A Persistent Svelte Store
+
+For cases when you need more than localized component state, Svelte offers _stores_. Svelte stores, however, do not persist data. We're going wrap the svelte `writable` store implementation with some Local Storage functionality.
+
+```js
+// localStorageStore.js
+import { writable, get } from "svelte/store";
+
```
Comment (dasDaniel) [inline components.md:538]: I don't see the localStorage assignment

```suggestion
const ls = localStorage
```
or update `ls` to `localStorage`

**Comment by dasDaniel** [comment]
I've also found a simple way to do a `localStorage` that uses a watch to update the store

```html
      <script>
        let lStore = {};
        let storeKey = "MY_STORE";

        try {
          lstore = JSON.parse(localStorage.getItem(storeKey));
        } catch (e) {}

        $: if (lStore) {
          localStorage.setItem(storeKey, JSON.stringify(lStore));
        }

        lStore.name = lStore.name || 'world';
      </script>

      <input bind:value="{lStore.name}" />
      <h1>Hello {lStore.name}!</h1>
```

File: components.md (hunk 2 above)
Comment (babycourageous) [reply components.md:538]: Ooof. Thanks for catching that.

**Comment by swyxio** [comment]
thanks @dasDaniel, i will add that in

**Pull request merged**